sunpy/map/mapbase.py

```diff
@@ -195,21 +196,26 @@ def __init__(self, data, header, plot_settings=None, **kwargs):
         self._validate_meta()
         self._shift = SpatialPair(0 * u.arcsec, 0 * u.arcsec)
 
-        if self.dtype == np.uint8:
-            norm = None
+        # Visualization attributes
```

Comment:
As discussed before, I think it is best to leave all the `cmap` and `norm` logic in the setters. However, I'll backtrack a bit and say that we should abandon the idea of using a property-setter approach for plot settings. Instead, I think the constructor should just look like

```python
def __init__(self, data, header, cmap='gray', norm=None, plot_settings=None, **kwargs):
    # Other stuff
    self.cmap = cmap
    self.norm = norm
    if plot_settings:
        self.plot_settings = plot_settings
        warnings.warn("Handling of plot_settings is deprecated. "
                                 "Pass the plot specific settings to either peek() or show()",
                                 DeprecationWarning)
        if 'norm' in self.plot_settings:
            self.norm = self.plot_settings['norm']
        if 'cmap' in self.plot_settings:
            self.cmap = self.plot_settings['cmap']
    else:
        self._plot_settings = {}
    # Other stuff
```